docs/self-hosting.md

```diff
@@ -0,0 +1,76 @@
+# Self-Hosting Guide for Discuit
+
+Welcome to the self-hosting guide for Discuit! Here, you'll find step-by-step instructions on how to set up and run Discuit on your own server.
+
+## Prerequisites
+
+Before you begin, make sure you have the following installed on your system:
+
+- [Go (1.21 or higher)](https://go.dev/)
+- [MariaDB](https://mariadb.org/)
+- [Redis](https://redis.io/)
+- [Node.js and NPM](https://nodejs.org/en/download/package-manager)
+- [Docker (optional, for running with Docker)](https://www.docker.com/products/docker-desktop/)
+- [libvips](https://libvips.github.io/libvips/install.html) (for image transformations)
+
+## 1. Setting Up the Development Environment
+
+### Create MariaDB Database
+
+```shell
+# Open MariaDB CLI
+mariadb -u root -p --binary-as-hex
+
+# Create a database named discuit (you may use a different name)
+create database discuit;
+
+# Exit MariaDB CLI
+exit;
+```
+
+## 2. Clone the Repository
+
+```shell
+git clone https://github.com/discuitnet/discuit.git && cd discuit
+```
+
+## 3. Configure Discuit
+
+Create a file named `config.yaml` in the root directory and copy the contents of `config.default.yaml` into it. Modify `config.yaml` with your desired configurations.
+
+## 4. Build and Run Discuit
+
+```shell
+./build.sh           # Build frontend and backend
+./discuit -migrate   # Run migrations
+./discuit -serve     # Start the server
+```
+
+## Running with Docker (Optional)
```

Comment:
Is this an optional 5th step, or a separately different way to build and deploy Discuit? If it is a 5th step, label it as such. If not, put the above steps into a separate heading and remove the `(Optional)` from this one.

Reply:
If this is a separate step, do the prerequisites still apply, or are they set up in the Dockerfile itself? If they don't apply, then you should move the prerequisites into the other section, and mention in this one that they don't need to be installed as they are set up automatically.

Reply:
Separated with their own prereqs.